aten/src/ATen/native/cuda/group_norm_kernel.cu

```diff
@@ -438,6 +682,55 @@ void GroupNormKernelImpl(
       });
 }
 
+template <typename T>
+void GroupNorm1dBackwardKernelImpl(
+    const TensorOptions& opt,
+    int64_t N,
+    int64_t C,
+    int64_t G,
+    const T* dY,
+    const T* X,
+    const T* mean,
+    const T* rstd,
+    const T* gamma,
+    T* dX,
+    T* dgamma,
+    T* dbeta,
+    acc_type<T, true>* c2,
+    acc_type<T, true>* c3) {
+  using T_ACC = acc_type<T, true>;
+  cudaStream_t cuda_stream = at::cuda::getCurrentCUDAStream();
+  if (dX != nullptr) {
+    TORCH_CHECK(c2 != nullptr);
+    TORCH_CHECK(c3 != nullptr);
+    const int64_t num_threads = (C / G) < cuda_utils::kCUDABlockReduceNumThreads
+        ? C10_WARP_SIZE
+        : cuda_utils::kCUDABlockReduceNumThreads;
+    Compute1dBackwardFusedParamsCUDAKernel<T>
+        <<<dim3(N, G), num_threads, 0, cuda_stream>>>(
+            C, G, dY, X, mean, rstd, gamma, c2, c3);
+    const int64_t B = (N * C + kCUDANumThreads - 1) / kCUDANumThreads;
+    GroupNorm1dBackwardCUDAKernel<T><<<B, kCUDANumThreads, 0, cuda_stream>>>(
+        N, C, G, dY, X, rstd, gamma, c2, c3, dX);
+    AT_CUDA_CHECK(cudaGetLastError());
+  }
+  if (dgamma != nullptr || dbeta != nullptr) {
+    if (N <= 128) {
+      const int64_t B = (C + kCUDANumThreads - 1) / kCUDANumThreads;
+      GammaBeta1dBackwardCUDAKernel1<T><<<B, kCUDANumThreads, 0, cuda_stream>>>(
+          N, C, G, dY, X, mean, rstd, dgamma, dbeta);
+    } else {
+      const int64_t B = (C + kReduceTileSize - 1) / kReduceTileSize;
+      constexpr int kThreadX = kReduceTileSize;
+      constexpr int kThreadY = kReduceTileSize / 2;
```

Comment:
please leave a comment here (or assert) to prevent someone from changing block dimensions, because otherwise shared memory sizes won't work.